yarn-project/stdlib/src/update-checker/update-checker.ts

```diff
@@ -0,0 +1,151 @@
+};
+
+type Config = {
+  baseURL: URL;
+  rollupVersion?: number | 'canonical';
+  nodeVersion?: string;
+  checkIntervalMs?: number;
+  registryContractAddress: EthAddress;
+  publicClient: ViemClient;
+  fetch?: typeof fetch;
+};
+
+export class UpdateChecker extends EventEmitter<EventMap> {
+  private runningPromise: RunningPromise;
+  private lastPatchedConfig: object = {};
+
+  constructor(
+    private configBasePath: URL,
+    private nodeVersion: string,
+    private rollupVersion: number | 'canonical',
+    private rollupAddressAtStart: EthAddress,
+    private fetch: typeof globalThis.fetch,
+    private getRollupAddress: (version: number | 'canonical') => Promise<EthAddress>,
+    private checkIntervalMs = 60_000, // every minute
+    private log = createLogger('foundation:update-check'),
+  ) {
+    super();
+    this.runningPromise = new RunningPromise(this.runChecks, this.log, this.checkIntervalMs);
+  }
+
+  public static async new(config: Config): Promise<UpdateChecker> {
+    const registryContract = new RegistryContract(config.publicClient, config.registryContractAddress);
+    const rollupAddress = await registryContract.getRollupAddress(config.rollupVersion ?? 'canonical');
+
+    return new UpdateChecker(
+      config.baseURL,
+      config.nodeVersion ?? getPackageVersion(),
+      config.rollupVersion ?? 'canonical',
+      rollupAddress,
+      config.fetch ?? fetch,
+      version => registryContract.getRollupAddress(version),
+      config.checkIntervalMs,
+    );
+  }
+
+  public async start() {
+    if (this.runningPromise.isRunning()) {
+      this.log.debug(`Can't start update checker again`);
+      return;
+    }
+
+    this.runningPromise.start();
+  }
+
+  public stop() {
+    if (!this.runningPromise.isRunning()) {
+      this.log.debug(`Can't stop update checker because it is not running`);
+      return;
+    }
+    return this.runningPromise.stop();
+  }
+
+  public trigger() {
+    return this.runningPromise.trigger();
+  }
+
+  private runChecks = async (): Promise<void> => {
+    await Promise.all([this.checkRollupVersion(), this.checkConfig()]);
+  };
+
+  private async checkRollupVersion(): Promise<void> {
+    try {
+      const canonicalRollup = await this.getRollupAddress(this.rollupVersion);
+      if (!canonicalRollup.equals(this.rollupAddressAtStart)) {
+        this.emit('newRollup', { currentRollup: this.rollupAddressAtStart, latestRollup: canonicalRollup });
+      }
+    } catch (err) {
+      this.log.warn(`Failed to check if there is a new rollup`, err);
+    }
+  }
+
+  private async checkConfig(): Promise<void> {
+    try {
+      const response = await this.fetch(
+        new URL('aztec-' + this.rollupAddressAtStart.toString() + '/index.json', this.configBasePath),
+      );
+      const body = await response.json();
+      if (!response.ok) {
+        if (response.status === 404) {
+          // no-op
+          return;
```

Comment:
Should we log this, so users get alerted of misconfigured auto update URLs? Or the idea is to not necessarily have an index.json for every rollup?